smacrobs_web/src/main/java/com/web/controller/MainController.java

```diff
@@ -65,22 +72,173 @@ public ModelAndView redirectFromFitbit(
 
         mv = new ModelAndView("dashboard");
-        mv.addObject("userProfile", userProfile);
+
         mv.addObject("sleepDetails", sleepDetails);
-        mv.addObject("heartRateDetails",heartRateDetails);
+        
+        mv = getHeartRate(mv, heartRateDetails);
+        mv = getActivity(mv,activityDetails);
+        mv = getFood(mv,foodDetails);
+        mv = getWater(mv,waterDetails);
+        mv = getUserProfile(mv, userProfile);
+        mv = getActivityGoals(mv, activityGoalDetails);
+        
+        
         return mv;
     }
 
-    private String getRedirectURL() {
+	private ModelAndView getActivityGoals(ModelAndView modelAndView, ActivityGoalDetails activityGoalDetails) {
+		String distance = "0";
+		String floor = "0";
+		String step = "0";
+		String cal = "0";
+		
+		try {
+			distance = activityGoalDetails.getGoals().getDistance();
+			floor = activityGoalDetails.getGoals().getFloors();
+			step = activityGoalDetails.getGoals().getSteps();
+			cal = activityGoalDetails.getGoals().getCaloriesOut();
+		}catch(Exception ex) {
+			logger.warn(ex.getMessage());
+		}
+		
+		modelAndView.addObject("distanceGoal", Float.parseFloat(distance));
+		modelAndView.addObject("floorsGoal", Integer.parseInt(floor));
+		modelAndView.addObject("stepsGoal", Integer.parseInt(step));
+		modelAndView.addObject("caloriesOutGoal", Integer.parseInt(cal));
+		System.out.println(modelAndView.toString());
+		return modelAndView;
+	}
+
+	private ModelAndView getUserProfile(ModelAndView modelAndView, UserProfile userProfile) {
+		String fullName = "0";
+		String gender = "0";
+		String age = "0";
+		String dateOfBirth = "0";
+		String weight = "0";
+		String weightUnit = "0";
+		double height = 0;
+		String heightUnit = "0";
+		String avatar = "0";
+		try {
+			fullName = userProfile.getUser().getFullName();
+			gender = userProfile.getUser().getGender();
+			age = userProfile.getUser().getAge();
+			dateOfBirth = userProfile.getUser().getDateOfBirth();
+			weight = userProfile.getUser().getWeight();
+			weightUnit = userProfile.getUser().getWeightUnit();
+			height = Math.round(Double.parseDouble(userProfile.getUser().getHeight()));
+			heightUnit = userProfile.getUser().getHeightUnit();
+			avatar = userProfile.getUser().getAvatar();
+		}catch(Exception ex) {
+			logger.warn(ex.getMessage());
+		}
+		
+		modelAndView.addObject("fullName", fullName);
+		modelAndView.addObject("gender", gender);
+		modelAndView.addObject("age", age);
+		modelAndView.addObject("dateOfBirth", dateOfBirth);
+		modelAndView.addObject("weight", weight);
+		modelAndView.addObject("weightUnit", weightUnit);
+		modelAndView.addObject("height", height);
+		modelAndView.addObject("heightUnit", heightUnit);
+		modelAndView.addObject("avatar", avatar);
+		System.out.println(modelAndView.toString());
+		return modelAndView;
+	}
+
+	private ModelAndView getWater(ModelAndView modelAndView, WaterDetails waterDetails) {
```

Comment:
@srinidhisrinivasaraghavan do we have all these getter methods here in controller or in service?